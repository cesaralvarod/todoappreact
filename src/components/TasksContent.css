.tasks-content {
  width: 100%;
}

@media (min-width: 700px) {
  .tasks-content {
    width: 60%;
  }
}
