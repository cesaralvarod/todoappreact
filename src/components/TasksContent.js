import React, { useContext } from "react";

// Components

import TasksCategory from "./TasksCategory";

// Context

import TaskContext from "../helpers/TaskContext";

// Styles

import "./TasksContent.css";

const TasksContent = () => {
  const { arrData } = useContext(TaskContext);
  const { data } = arrData;

  let tasksUncomplete = data.filter((task) => {
    if (!task.done) {
      return task;
    }
  });

  let tasksComplete = data.filter((task) => {
    if (task.done) {
      return task;
    }
  });

  const renderCompleteTasks = () => {
    if (tasksComplete.length > 0) {
      return <TasksCategory title="Complete Tasks" tasks={tasksComplete} />;
    }
  };

  return (
    <div className="tasks-content">
      <TasksCategory tasks={tasksUncomplete} />
      {renderCompleteTasks()}
    </div>
  );
};

export default TasksContent;
