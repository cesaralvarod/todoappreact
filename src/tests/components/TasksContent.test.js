import React from "react";
import { shallow } from "enzyme";
import TasksContent from "../../components/TasksContent";
import TaskContext from "../../helpers/TaskContext";
import { demoData } from "../fixtures/demoData";

describe("Testing in component TasksContent", () => {
  const TestComponent = () => (
    <TaskContext.Provider value={{ arrData: demoData }}>
      <TasksContent />
    </TaskContext.Provider>
  );

  const wrapper = shallow(<TestComponent />);

  test("Component TasksContent should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
