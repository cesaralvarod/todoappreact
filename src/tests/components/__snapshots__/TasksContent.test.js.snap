// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component TasksContent Component TasksContent should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "arrData": Object {
        "data": Array [
          Object {
            "content": "Learn react",
            "done": false,
            "id": 0,
          },
          Object {
            "content": "Do homework",
            "done": true,
            "id": 1,
          },
          Object {
            "content": "Watch tv",
            "done": false,
            "id": 2,
          },
        ],
      },
    }
  }
>
  <TasksContent />
</ContextProvider>
`;
